### PR TITLE
[risk=low][no ticket] Fix FetchWorkspaceDetails tool

### PR DESCRIPTION
**Pull request by jmthibault79**
I'm not sure how useful this tool remains, but I tried it out while looking into an issue and I found it didn't work.  This PR restores it to functionality plus some minor improvements to bring it more in line with the current codebase.

Tested by running 
```
./project.rb fetch-workspace-details --project all-of-us-workbench-test --workspace-namespace aou-rw-test-d15fd8ee
```

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: api/libproject/devstart.rb [hunk 1]
```diff
@@ -1425,10 +1425,10 @@ def fetch_workspace_details(cmd_name, *args)
   op.opts.project = TEST_PROJECT
 
   op.add_typed_option(
-      "--workspace-project-id [workspace-project-id]",
```
Comment (jmthibault79) [inline api/libproject/devstart.rb:1428]: "project id" is an ambiguous name.  this is NOT the google project.

File: api/tools/src/main/java/org/pmiops/workbench/tools/FetchWorkspaceDetails.java [hunk 2]
```diff
@@ -26,7 +30,14 @@
  * workspaces found.
  */
 @Configuration
-@Import({FireCloudServiceImpl.class, FireCloudConfig.class})
+@Import({
```
Comment (jmthibault79) [inline api/tools/src/main/java/org/pmiops/workbench/tools/FetchWorkspaceDetails.java:33]: all of these imports are necessary.  unclear why that would have been different in the past.

File: api/tools/src/main/java/org/pmiops/workbench/tools/FetchWorkspaceDetails.java [hunk 3]
```diff
@@ -39,60 +50,72 @@ public class FetchWorkspaceDetails extends Tool {
...
-        sb.append("Workspace Namespace: " + workspace.getWorkspaceNamespace() + "\n");
-        sb.append("Creator: " + workspace.getCreator().getUsername() + "\n");
-        sb.append("GCS bucket path: gs://" + bucketName + "\n");
-        sb.append("Collaborators:\n");
-        for (Map.Entry<String, RawlsWorkspaceAccessEntry> aclEntry : acl.entrySet()) {
-          sb.append("\t" + aclEntry.getKey() + " (" + aclEntry.getValue().getAccessLevel() + ")\n");
-        }
-
-        sb.append(String.join("\n", Collections.nCopies(3, "***")));
+      // project.rb swallows exceptions, so we need to catch and log them here
```
Comment (jmthibault79) [inline api/tools/src/main/java/org/pmiops/workbench/tools/FetchWorkspaceDetails.java:67]: until we replace project.rb, we need workarounds like this

File: api/tools/src/main/java/org/pmiops/workbench/tools/FetchWorkspaceDetails.java [hunk 4]
```diff
@@ -39,60 +50,72 @@ public class FetchWorkspaceDetails extends Tool {
...
+      } catch (Exception e) {
+        log.severe("Error fetching workspace details: " + e.getMessage());
+        e.printStackTrace();
       }
-
-      log.info(sb.toString());
     };
   }
 
+  public void getDetails(
```
Comment (jmthibault79) [inline api/tools/src/main/java/org/pmiops/workbench/tools/FetchWorkspaceDetails.java:77]: this is just the main contents of `run()` from before

File: api/tools/src/main/java/org/pmiops/workbench/tools/FetchWorkspaceDetails.java [hunk 5]
```diff
@@ -39,60 +53,83 @@ public class FetchWorkspaceDetails extends Tool {
...
+
+      String bucketName =
+          fireCloudService
+              .getWorkspaceAsService(workspaceNamespace, workspace.getFirecloudName())
+              .getWorkspace()
+              .getBucketName();
+
+      sb.append("\nWorkspace Name: ").append(workspace.getName()).append("\n");
+      sb.append("Workspace Namespace: ").append(workspace.getWorkspaceNamespace()).append("\n");
+      sb.append("Google Project: ").append(workspace.getGoogleProject()).append("\n");
```
Comment (jmthibault79) [inline api/tools/src/main/java/org/pmiops/workbench/tools/FetchWorkspaceDetails.java:109]: added a few fields like this one

**Review by Qi77Qi — APPROVED** [review]
thanks for doing this!

**Pull request merged**